pyvista/utilities/reader.py

```diff
@@ -980,6 +983,185 @@ class Plot3DMetaReader(BaseReader):
+
+    """
+
+    _class_reader = staticmethod(_vtk.lazy_vtkCGNSReader)
+
+    @property
+    def distrubte_blocks(self) -> bool:
+        """Distrubte each block in each zone across ranks.
+
+        To make the reader disregard piece request and read all blocks in the
+        zone, set this to ``False``. The default is ``True``.
+
+        Returns
+        -------
+        bool
+            If ``True``, distrubte each block in each zone across ranks.
+
+        Examples
+        --------
+        Disable distributing blocks.
+
+        >>> import pyvista
+        >>> from pyvista import examples
+        >>> filename = examples.download_cgns_multi(load=False)
+        >>> reader = pyvista.get_reader(filename)
+        >>> reader.distrubte_blocks = False
+        >>> reader.distrubte_blocks
+        False
+
+        """
+        return bool(self._reader.GetDistributeBlocks())
+
+    @distrubte_blocks.setter
+    def distrubte_blocks(self, value):
+        self._reader.SetDistributeBlocks(value)
+
+    def enable_all_bases(self):
+        """Enable reading all bases.
+
+        By default only the 0th base is read.
+
+        Examples
+        --------
+        Read all bases
+
+        >>> import pyvista
+        >>> from pyvista import examples
+        >>> filename = examples.download_cgns_multi(load=False)
+        >>> reader = pyvista.get_reader(filename)
+        >>> reader.enable_all_bases()
+
+        """
+        self._reader.EnableAllBases()
+
+    def disable_all_bases(self):
+        """Disable reading all bases.
+
+        By default only the 0th base is read.
+
+        Examples
+        --------
+        Disable reading all bases.
+
+        >>> import pyvista
+        >>> from pyvista import examples
+        >>> filename = examples.download_cgns_multi(load=False)
+        >>> reader = pyvista.get_reader(filename)
+        >>> reader.disable_all_bases()
+
+        """
+        self._reader.DisableAllBases()
+
+    def enable_all_point_arrays(self):
+        """Enable reading all point arrays.
+
+        By default none are loaded.
+
+        Examples
+        --------
+        Read all point arrays.
+
+        >>> import pyvista
+        >>> from pyvista import examples
+        >>> filename = examples.download_cgns_multi(load=False)
+        >>> reader = pyvista.get_reader(filename)
+        >>> reader.enable_all_point_arrays()
+
+        """
+        self._reader.EnableAllPointArrays()
+
+    def disable_all_point_arrays(self):
+        """Disable reading all point arrays.
+
+        This is the default behavior.
+
+        Examples
+        --------
+        Disable reading all point arrays.
+
+        >>> import pyvista
+        >>> from pyvista import examples
+        >>> filename = examples.download_cgns_multi(load=False)
+        >>> reader = pyvista.get_reader(filename)
+        >>> reader.disable_all_point_arrays()
+
+        """
+        self._reader.DisableAllPointArrays()
+
+    def enable_all_cell_arrays(self):
```

Comment:
I would add an `__init__` method that enables all arrays by default.  Then we can document that this is different than vtk default.

Reply:
I agree with changing the default behavior and I've done so in https://github.com/pyvista/pyvista/pull/2366/commits/893bb803cf5a868e540cfce2d8c87c9198e47500.